### PR TITLE
[WPI] Feature tests form

### PR DESCRIPTION
**Pull request by plepaisant**
### Description

Tests created  for state, props, reference lists, store and definitions related to the form component

Please tell me the tests to be done

### CODE REVIEW

**Comment by Bernardstanislas** [comment]
@plepaisant please refer to [this blog](http://www.zsoltnagy.eu/writing-automated-tests-with-mocha-and-chai/) in order to write accurate mocha test descriptions :)
thank you !

File: src/common/form/__tests__/fake-form.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    mixins: [formMixin],
+    referenceNames :['papas', 'singe'],
+    action :action,
+    stores: [{
+        store: contactStore,
+        properties: ['contact', 'commandes']
+    }],
+
+    componentWillMount() {
+        this._htmlId = uuid.v4();
```
Comment (pierr) [inline src/common/form/__tests__/fake-form.js]: use lodash uniqueId

File: src/common/form/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+const isEmpty = require("lodash/lang/isEmpty");
+const isEqual = require('lodash/lang/isEqual');
+const keys = require('lodash/object/keys');
+const difference = require('lodash/array/difference');
+
+initEnvironment();
+
+describe.only('The Form component', () => {
+
+    describe('Basic Form with no action, not editable ', () => {
```
Comment (pierr) [inline src/common/form/__tests__/index.js]: when the form has no action and isn't ediable

File: src/common/form/__tests__/index.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        });
+
+        it('definitions are correct', () => {
+            expect(isEmpty(reactComponent.definition)).to.equal(false);
+            expect(isEmpty(difference(reactComponent,entities.contact)));
+            expect(isEmpty(difference(entities.contact,reactComponent)));
+        });
+    });
+
+    describe('Form with action, editable ', () => {
```
Comment (pierr) [inline src/common/form/__tests__/index.js]: when the form ....

File: src/common/form/__tests__/index.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        });
+
+        it('props are correct', () => {
+            expect(reactComponent.props.hasEdit).to.equal(true);
+            expect(reactComponent.props.hasForm).to.equal(true);
+            expect(reactComponent.props.hasDelete).to.equal(false);
+
+        });
+
+        it('store is correct', () => {
```
Comment (pierr) [inline src/common/form/__tests__/index.js]: should have a correct store

**Comment by plepaisant** [comment]
tests description are more accurate

**Pull request closed without merging**